framework/src/media/audio/audio_manager.c

```diff
@@ -1079,31 +1107,25 @@ int get_output_audio_volume(void)
 
 	close(fd);
 
-	return (int)g_audio_out_cards[g_actual_audio_out_card_id].config.volume;
-#endif
-	return 0;
+	*volume = g_audio_out_cards[g_actual_audio_out_card_id].config.volume;
+
+	return AUDIO_MANAGER_SUCCESS;
 }
 
-audio_manager_result_t set_input_audio_volume(uint8_t volume)
+audio_manager_result_t set_input_audio_gain(uint8_t gain)
 {
-#ifndef CONFIG_AUDIO_EXCLUDE_VOLUME
-	if (g_audio_in_cards[g_actual_audio_in_card_id].config.volume == volume) {
-		return AUDIO_MANAGER_SUCCESS;
+	if (g_audio_in_cards[g_actual_audio_in_card_id].config.max_volume < 0) {
```

Comment:
max_volume datatype is unsigned.
This logic is never called.

Reply:
max_volume is changed into 'int' to cover the case that CONFIG_AUDIO_EXCLUDE_XXX is set.